crates/trippy/src/main.rs

```diff
@@ -485,21 +485,29 @@ mod tests {
     use super::*;
     use test_case::test_case;
 
-    #[test_case(include_str!("../test_resources/tui/tui_theme_items.txt"), &tui_theme_items(); "tui theme items match")]
-    #[test_case(include_str!("../test_resources/tui/tui_binding_commands.txt"), &tui_binding_commands(); "tui binding commands match")]
-    #[test_case(include_str!("../test_resources/config/completions_bash.txt"), &shell_completions(Shell::Bash).unwrap(); "generate bash shell completions")]
-    #[test_case(include_str!("../test_resources/config/completions_elvish.txt"), &shell_completions(Shell::Elvish).unwrap(); "generate elvish shell completions")]
-    #[test_case(include_str!("../test_resources/config/completions_fish.txt"), &shell_completions(Shell::Fish).unwrap(); "generate fish shell completions")]
-    #[test_case(include_str!("../test_resources/config/completions_powershell.txt"), &shell_completions(Shell::PowerShell).unwrap(); "generate powershell shell completions")]
-    #[test_case(include_str!("../test_resources/config/completions_zsh.txt"), &shell_completions(Shell::Zsh).unwrap(); "generate zsh shell completions")]
-    #[test_case(include_str!("../test_resources/config/trip.1"), &man_page().unwrap(); "generate man page")]
-    fn test_output(expected: &str, actual: &str) {
-        if remove_whitespace(actual.to_string()) != remove_whitespace(expected.to_string()) {
-            pretty_assertions::assert_eq!(actual, expected);
-        }
+    #[test_case(&tui_theme_items(), "tui theme items match"; "tui theme items match")]
+    #[test_case(&tui_binding_commands(), "tui binding commands match"; "tui binding commands match")]
+    #[test_case(&shell_completions(Shell::Bash).unwrap(), "generate bash shell completions"; "generate bash shell completions")]
+    #[test_case(&shell_completions(Shell::Elvish).unwrap(), "generate elvish shell completions"; "generate elvish shell completions")]
+    #[test_case(&shell_completions(Shell::Fish).unwrap(), "generate fish shell completions"; "generate fish shell completions")]
+    #[test_case(&shell_completions(Shell::PowerShell).unwrap(), "generate powershell shell completions"; "generate powershell shell completions")]
+    #[test_case(&shell_completions(Shell::Zsh).unwrap(), "generate zsh shell completions"; "generate zsh shell completions")]
+    #[test_case(&man_page().unwrap(), "generate man page"; "generate man page")]
+    fn test_output(actual: &str, name: &str) {
+        insta(name, || {
+            insta::assert_snapshot!(remove_whitespace(actual.to_string()));
+        });
+    }
+
+    pub fn insta<F: FnOnce()>(name: &str, f: F) {
+        let mut settings = insta::Settings::new();
+        settings.set_snapshot_suffix(name.replace(' ', "_"));
+        settings.set_snapshot_path("../tests/resources/snapshots");
```

Comment:
I do have some reservations regarding the relative path use here but so far haven't seen it cause a problem. Did you dislike the absolute path approach?

Reply:
Since it's only called from one place I think it's fine as all of the users pass their use case as a closure.